lib/controller.js

```diff
@@ -10,255 +10,260 @@ var updates = require('./update-fetch');
-        seeker.removeListener('end', searchComplete);
-        // Stop searching
-        seeker.stop();
-        // Send this Tessel back to the caller
-        logAndFinish(tessel);
-      }
-      // Otherwise
-      else {
-        // Store this Tessel with the others
-        tessels.push(tessel);
-      }
-    });
-
-    seeker.once('end', searchComplete);
-
-    // Accesses `tessels` in closure
-    function logAndFinish(tessel) {
-      // The Tessels that we won't be using should have their connections closed
-      var connectionsToClose = tessels;
+          }
+        }
 
-      if (tessel) {
-        logs.info(sprintf('Connected to %s over %s', tessel.name, tessel.connection.connectionType));
-        connectionsToClose.splice(tessels.indexOf(tessel), 1);
+        // When we find Tessels
+        seeker.on('tessel', function(tessel) {
+          // Check if this name matches the provided option (if any)
+          // This speeds up development by immediately ending the search
+          if (opts.name && opts.name === tessel.name) {
+            // Remove this listener because we don't need to search for the Tessel
+            seeker.removeListener('end', searchComplete);
+            // Stop searching
+            seeker.stop();
+            // Send this Tessel back to the caller
+            logAndFinish(tessel);
+          }
+          // Otherwise
+          else {
+            // Store this Tessel with the others
+            tessels.push(tessel);
+          }
+        });
 
-        controller.closeTesselConnections(connectionsToClose)
-          .then(function() {
-            return resolve(tessel);
-          });
-      } else {
-        logs.info('Please specify a Tessel by name [--name <tessel name>]');
-        controller.closeTesselConnections(connectionsToClose)
-          .then(function() {
-            reject('Multiple possible Tessel connections found.');
-          });
-      }
-    }
-  });
+        seeker.once('end', searchComplete);
+
+        // Accesses `tessels` in closure
+        function logAndFinish(tessel) {
+          // The Tessels that we won't be using should have their connections closed
+          var connectionsToClose = tessels;
+          if (tessel) {
+            logs.info(sprintf('Connected to %s over %s', tessel.name, tessel.connection.connectionType));
+            connectionsToClose.splice(tessels.indexOf(tessel), 1);
+            controller.closeTesselConnections(connectionsToClose)
+              .then(function() {
+                return resolve(tessel);
+              });
+          } else {
+            logs.info('Please specify a Tessel by name [--name <tessel name>]');
+            controller.closeTesselConnections(connectionsToClose)
+              .then(function() {
+                return reject('Multiple possible Tessel connections found.');
+              });
+          }
+        }
+      });
+    });
 };
 
 /*
```

Comment:
Can you give me a run down for all the changes here? I'm not sure I understand all of this.